### ISSUE TITLE
Dictionary tag breaks when custom option is not set

### ISSUE THREAD

**Issue by bynicolas** [statamic/cms#11004]
### Bug description

The new dictionary antler tag introduced in v5.31.0 (#10885) throws a not found exception when a custom option is not set.

I have added a custom `multiple` toggle option to my dictionary. When the option is not set, the page with the form using that dictionary throws a not found exception.

### How to reproduce

in the dictionary definition class, add a custom field option item:

```php
protected function fieldItems()
{
    return [
        'multiple' => [
            'type' => 'toggle',
            'value' => false,
            'instructions' => 'Allow multiple selections.'
        ]
    ];
}
```

Use the dictionary tag in your form field. I use a form partial defined in `resources/views/partials/forms/fields/dictionary.antlers.html` dynamically rendered in my form template.

```antlers
<select name="{{ handle }}{{ dictionary:multiple ?= " []" }}"  ...>
...
</select>
```

If multiple is not define because the toggle option on the form field is set to disabled, then the exception is thrown

### Environment

```yaml
Environment
Application Name: Dev site
Laravel Version: 11.28.1
PHP Version: 8.3.12
Composer Version: 2.7.7
Environment: local
Debug Mode: ENABLED
URL: cliniqueia.test
Maintenance Mode: OFF
Timezone: America/Toronto
Locale: fr_CA

Cache
Config: NOT CACHED
Events: NOT CACHED
Routes: NOT CACHED
Views: NOT CACHED

Drivers
Broadcasting: log
Cache: file
Database: sqlite
Logs: stack / single
Mail: log
Queue: sync
Session: file

Statamic
Addons: 2
Sites: 2 (Dev site - en)
Stache Watcher: Enabled
Static Caching: Disabled
Version: 5.31.0 PRO

Statamic Addons
alt-design/alt-redirect: 1.3.2
bynicolas/ignite-seo: dev-main
```

### Installation

Fresh statamic/statamic site via CLI

**Comment by ryanmitchell** [comment]
`{{ dictionary:multiple }}` would be looking for a dictionary called multiple - which doesn't seem right from what you've said?

Should you not just be looking for the `multiple` variable?

**Comment by duncanmcclean** [comment]
Just to confirm, you're using this in the context of a form field and you're trying to pass a config option to the dictionary?

**Comment by bynicolas** [comment]
Yes @duncanmcclean, this is used in the context of a form field. The issue is arising in the frontend template when the config option is not set because the toggle is off (false). The CP is working fine and no problems in adding the the config option to the dictionary

Sorry for not being totally clear, hope this clarifies it a bit more.

**Comment by jasonvarga** [comment]
Why are you adding your own `multiple` field? That's already part of the fieldtype. The "max items" field. As long as you don't set it to `1`, it'll allow you to pick multiple.

**Comment by bynicolas** [comment]
It was to be consistent with the way select field types are rendered. I am using this option to generate the appropriate template. I am not using the template rendered by default. 

Regardless, I think this is a bug introduced since it crashes when rendering the antler tag and the key is not present. Normally antlers would gracefully and silently handle these cases.

In other words, regardless of my use case here, another custom option added to a dictionary field would likely cause the same issue if the field is not set (not tested but I could if you want) and it makes the behaviour expected from antlers inconsistent here. It should handle the exception when a key is not found like it does normally. Well in `v5.30.0` and earlier it was working fine.

**Comment by Upending36** [comment]
Thank you.


Sent from the all new AOL app for iOS


On Friday, October 25, 2024, 10:46 PM, bynicolas ***@***.***> wrote:




It was to be consistent with the way select field types are rendered. I am using this option to generate the appropriate template. I am not using the template rendered by default.

Regardless, I think this is a bug introduced since it crashes when rendering the antler tag and the key is not present. Normally antlers would gracefully and silently handle these cases.

In other words, regardless of my use case here, another custom option added to a dictionary field would likely cause the same issue if the field is not set (not tested but I could if you want) and it makes the behaviour expected from antlers inconsistent here. It should handle the exception when a key is not found like it does normally. Well in v5.30.0 and earlier it was working fine.

**Comment by ryanmitchell** [comment]
Did you miss my reply? You are calling dictionary:multiple which will be looking for a dictionary called multiple - which is where the error comes from. That isn’t the name of your variable surely?

**Comment by jasonvarga** [comment]
I see your issue.

The actual _config_ of the field has a `dictionary` key and you want the `multiple` field inside.

Since you are doing something pretty custom here, I would say just tell Antlers you want the variable and not the tag.

```antlers
{{ $dictionary:multiple }}
```
